pkg/reconciler/ingress/ingress.go

```diff
@@ -249,38 +247,37 @@ func (r *Reconciler) reconcileCertSecrets(ctx context.Context, ia *v1alpha1.Ingr
 }
 
 func (r *Reconciler) reconcileVirtualServices(ctx context.Context, ia *v1alpha1.Ingress,
-	desired []*v1alpha3.VirtualService) error {
+	desired map[string]*v1alpha3.VirtualService) error {
+	// Sort VirtualService names for a consistent processing order.
+	names := sets.NewString()
```

Comment:
This is very heavyhanded (you also call `List` twice which means, sorting twice).
Wouldn't just collecting the, into a `[]string` and then calling `sort.Slice` much better?
you should not have duplicates here. I think even sorting the list that we had originally is the best way.

Reply:
Done